src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -142,16 +142,11 @@ struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
             auto ignore_index_val =
                 parser.parse_value(info.attributes.at("ignore_index")).at<int64_t>();
 
-            if(ignore_index_val >= input_classes or abs(ignore_index_val) > input_classes)
+            if(ignore_index_val >= input_classes or ignore_index_val < 0)
```

Comment:
Pardon me, this thing is likely falling through the cracks. I meant that any index value that is negative is to be considered like the out of bounds. But that doesn't imply that we should ignore it. These `ignore_index` OOB values are still to be processed/compared when looking at `l[i][d1][d2]..[dk]`. As per this specification:

> l[i][d1][d2]...[dk]  = 0, when labels[n][d1][d2]...[dk] = ignore_index

Reply:
Yup. Handled this. 

What occurs is the ignore index is added to the weights but if its out of bounds, it doesn't make it as part of the weight vector, which is always multiplied at the end once softmax + log + neg is applied to the selected score data. 

If ignore index is out of bounds, then we never update our weight vector with a zero to cancel out the score value before its set for the reduction step

Reply:
I think this is a bit more complicated than realized. Likely will need to modify parser again since we can't read label on parse but have the ignore index of the operator. Will need to add some additional masking.

Reply:
Solved. Get equivalent output with using a bitmask and where as we can use the input index and check with equals/where after the loss is calculated. 

Verified the results with Onnxruntime and used that as the gold data in the added verify tests.